api/app.go

```diff
@@ -99,6 +126,35 @@ func (app *App) Configure() error {
 		return err
 	}
 
+	//we are customizing the default http error reply
+	runtime.HTTPError = func(ctx context.Context, mux *runtime.ServeMux, marshaler runtime.Marshaler,
+		w http.ResponseWriter, _ *http.Request, err error) {
+		w.Header().Set("Content-Type", marshaler.ContentType())
+		var s int
+
+		st, s := app.getStatusCodeFromError(err)
+
+		w.WriteHeader(s)
+
+		type errorBody struct {
+			Success bool   `json:"success"`
+			Reason  string `json:"reason"`
+		}
+
+		body := &errorBody{
+			Success: false,
+			Reason:  st.Message(),
+		}
+
+		buf, merr := marshaler.Marshal(body)
+		if merr != nil {
+			app.Logger.Error("Failed to marshal error body,", zap.Error(merr))
+		}
+		if _, err := w.Write(buf); err != nil {
```

Comment:
Don't write the  buffer if you failed to Marshal, it's probably nil in this case and we'll likely panic. I suggest you return in the if above (`if merr != nil`).